Live message status updates

tmaybe opened the pull request:
Adds live message status updates to message pages.

Closes #3

tmaybe committed:
use Postgres adapter for ActionCable in prod & dev

Instead of redis; at the suggestion of @bensheldon and @daguar, to
reduce moving parts/points of failure.

tmaybe committed:
failing test for message ids

tmaybe committed:
add unique id to message cards

tmaybe committed:
sending message status updates through the cable

tdooner commented on app/jobs/message_broadcast_job.rb:
```diff
@@ -1,10 +1,10 @@
   queue_as :default
 
-  def perform(message)
+  def perform(message, dom_id)
```
Maybe since this method 1) receives the message object, 2) is already responsible for converting it to HTML, and 3) is also responsible for the `ActionCable.server.broadcast` call, it makes sense for this method to also be responsible for calculating the `message_dom_id`.

This would make the call sites a little simpler (and in general prevent changes to the method signature of `perform` here), which seems like a good win for testability.

tmaybe replied:
Good insight, thank you!

tdooner reviewed (approved):
A bit of explanation of what motivates the `appendOrReplace` would be helpful - I assume it's due to some kind of race where the same message can be appended multiple times?

tmaybe committed:
calculate dom_id in the broadcast job

tmaybe committed:
be more explicit about new vs. updated messages

appendOrReplace was supposed to handle both new messages and status
updates to existing messages, but @tdooner's comment made me realize
that there could be a case where messages receiving updates aren't in
the DOM and are mistakenly appended instead of updated. To avoid this,
I'm making it explicit whether I'm broadcasting a new message vs an
update, and I don't do anything with an update if the matching element
isn't in the DOM.

tmaybe committed:
gem update, including removing redis (for now)

tmaybe committed:
ApplicationJob testing
